pkg/aws/aws.go

```diff
@@ -87,41 +93,55 @@ var (
-	for _, service := range services {
-		switch service {
+	// There are two scenarios:
+	// 1. Running locally, the user must pass in a region and profile to use
+	// 2. Running within an EC2 instance and the region and profile can be derived
+	// I'm going to use the AWS SDK to handle this for me. If the user has provided a region and profile, it will use that.
+	// If not, it will use the EC2 instance metadata service to determine the region and credentials.
+	// This is the same logic that the AWS CLI uses, so it should be fine.
+	options := []func(*awsconfig.LoadOptions) error{awsconfig.WithEC2IMDSRegion()}
+	if config.Region != "" {
+		options = append(options, awsconfig.WithRegion(config.Region))
+	}
+	if config.Profile != "" {
+		options = append(options, awsconfig.WithSharedConfigProfile(config.Profile))
+	}
+	ac, err := awsconfig.LoadDefaultConfig(context.Background(), options...)
+	if err != nil {
+		return nil, err
+	}
+	for _, service := range config.Services {
+		switch strings.ToUpper(service) {
 		case "S3":
-			// There are two scenarios:
-			// 1. Running locally, the user must pass in a region and profile to use
-			// 2. Running within an EC2 instance and the region and profile can be derived
-			// I'm going to use the AWS SDK to handle this for me. If the user has provided a region and profile, it will use that.
-			// If not, it will use the EC2 instance metadata service to determine the region and credentials.
-			// This is the same logic that the AWS CLI uses, so it should be fine.
-			options := []func(*awsconfig.LoadOptions) error{awsconfig.WithEC2IMDSRegion()}
-			if config.Region != "" {
-				options = append(options, awsconfig.WithRegion(config.Region))
-			}
-			if config.Profile != "" {
-				options = append(options, awsconfig.WithSharedConfigProfile(config.Profile))
-			}
-			ac, err := awsconfig.LoadDefaultConfig(context.Background(), options...)
-			if err != nil {
-				return nil, err
-			}
-
 			client := costexplorer.NewFromConfig(ac)
 			collector, err := s3.New(config.ScrapeInterval, client)
 			if err != nil {
 				return nil, fmt.Errorf("error creating s3 collector: %w", err)
 			}
 			collectors = append(collectors, collector)
+		case "EKS":
+			pricingService := pricing.NewFromConfig(ac)
+			computeService := ec2.NewFromConfig(ac)
+			regions, err := computeService.DescribeRegions(context.TODO(), &ec2.DescribeRegionsInput{AllRegions: aws.Bool(false)})
+			if err != nil {
+				return nil, fmt.Errorf("error getting regions: %w", err)
+			}
+			regionClientMap := make(map[string]ec2client.EC2)
+			for _, r := range regions.Regions {
+				client, err := newEc2Client(*r.RegionName, config.Profile)
+				if err != nil {
+					return nil, fmt.Errorf("error creating ec2 client: %w", err)
+				}
+				regionClientMap[*r.RegionName] = client
+			}
+			collector := eks.NewCollector(config.Region, config.Profile, config.ScrapeInterval, pricingService, computeService, regions.Regions, regionClientMap)
```

Comment:
> One other super NIT is that the s3 New() function returns a checked error, but it's always nil, can we remove that error check? Just so we have a consisted interface for the "New Collector" function

💯 I was considering making the change to S3 to no longer return an error. That's from a previous times where `s3.New` creating the clients within the method instead of having them passed in.